test/coverage-test/testing-options.mjs

```diff
@@ -194,6 +194,13 @@ const testCases = [
       include: ['coverage-report-tests/merge-reports.test.ts'],
     },
   },
+  {
+    testConfig: {
+      name: 'thresholds autoUpdate',
+      include: ['option-tests/thresholds-auto-update.test.ts'],
+      coverage: { enabled: false },
+    },
+  },
```

Comment:
These `thresholds-auto-update.test.ts` tests are not really "option tests", but this was the easiest way to add actual unit tests for this. Structure of coverage tests need some major refactoring like other tests did in https://github.com/vitest-dev/vitest/pull/5600. I'll leave that refactoring out of this PR though.